src/hpc/thread.c

```diff
@@ -191,7 +213,19 @@ static NOINLINE void GrowStack(void)
 
 static NOINLINE void SetupTLS(void)
 {
-#ifndef HAVE_NATIVE_TLS
+#if defined USE_MACOS_PTHREAD_TLS_ASM
+    static char asm_code[] = {
+        // movq %gs:(,%rdi,8), %rax
+        // retq
+        0x65, 0x48, 0x8b, 0x04, 0xfd, 0x00, 0x00, 0x00, 0x00, 0xc3
+    };
+    if (memcmp((void *)pthread_getspecific, asm_code, sizeof(asm_code)) != 0) {
```

Comment:
```suggestion
    if (memcmp(pthread_getspecific, asm_code, sizeof(asm_code)) != 0) {
```

Reply:
That code is actually now unnecessary that it's been moved to the configure script. That said, I remember it giving me a warning so I would rather leave the cast in there.

Reply:
It might give a warning in C++, but never in C.